### PR TITLE
Use secret key refs instead of plain secret values

### PR DESCRIPTION
**Pull request by thekad**
```release-note
NONE
```

Some objects make use of environment variables that come from secret values, but these values end up logged by the sync pod.
Sync pod doesn't log the values when the object type is Secret, so by moving these values to a Secret and using a key ref we avoid logging secrets to the sync pod logs.

### CODE REVIEW

**Comment by thekad** [comment]
still wip because i haven't been able to test this by running the sync pod, as soon as i can verify this works will remove the wip tag

File: pkg/sync/v4/translate.go [hunk 1]
```diff
@@ -635,12 +639,7 @@ var translations = map[string][]struct {
 			},
 		},
 	},
-	"Service/default/router-stats": {
-		{
-			Path:     jsonpath.MustCompile("$.metadata.annotations['prometheus.openshift.io/password']"),
-			Template: "{{ .Config.RouterStatsPassword }}",
-		},
-	},
+	"Service/default/router-stats": {},
```
Comment (jim-minter) [inline pkg/sync/v4/translate.go]: remove this line

**Comment by jim-minter** [comment]
one nit, then please squash, then lgtm

**Comment by thekad** [comment]
/hold

not sure what is going here, gonna check

**Comment by jim-minter** [comment]
/lgtm cancel

cluster isn't coming up because registry isn't coming up - something wrong with the PR I guess

File: pkg/sync/v4/translate.go [hunk 2]
```diff
@@ -544,6 +536,18 @@ var translations = map[string][]struct {
 			Template: "{{ String (PrivateKeyAsBytes .Config.Certificates.Router.Key) }}",
 		},
 	},
+	"Secret/default/docker-registry-http": {
+		{
+			Path:     jsonpath.MustCompile("$.stringData.password"),
+			Template: "{{ .Config.RegistryHTTPSecret }}",
```
Comment (jim-minter) [inline pkg/sync/v4/translate.go]: Missing Base64Encode here I think

Comment (thekad) [reply pkg/sync/v4/translate.go]: No, this is going to `stringData` not `data`

**Comment by jim-minter** [comment]
merging this
@thekad please make sure there is a follow-up issue open to get these stats into Geneva - work with @Makdaam to achieve this

**Pull request merged**